app/src/main/kotlin/com/softteco/template/data/profile/repository/ProfileRepositoryImpl.kt

```diff
@@ -174,10 +103,22 @@ internal class ProfileRepositoryImpl @Inject constructor(
     }
 
     @Suppress("TooGenericExceptionCaught")
-    override suspend fun logout(): Result<Unit> {
+    override suspend fun updateUser(updateUserDto: UpdateUserDto): Result<Profile> {
         return try {
-            authTokenEncryptedDataStore.updateData { token -> token.copy(token = "") }
-            Result.Success(Unit)
+            val result = requestWithRetry(delay = REQUEST_RETRY_DELAY) {
+                profileApi.updateUser(updateUserDto)
+            }
+            when (result) {
+                is ApiError -> Result.Error(AppError.AuthError.findByCode(result.errorBody?.code))
+                is ApiException -> Result.Error(AppError.NetworkError())
+                is ApiSuccess -> {
+                    val profile = result.data.toModel()
+                    when (val cacheResult = cacheProfile(profile)) {
+                        is Result.Error -> Result.Error(cacheResult.error)
```

Comment:
I'm not sure if an error should be returned in this case. The record was changed, the updated model was returned. It looks like the error is our internal problem (we need to log it), and there is nothing stopping us from showing the data to the user

Reply:
In general, I agree.  
Removed handling caching result at all here, because error we log inside `cacheProfile` function.